node/src/gql/mutation/me/update-subject.ts

```diff
@@ -57,53 +100,34 @@ export const updateSubject = {
     context: { user: User }
   ): Promise<Subject> => {
     const subjectUpdate = args.subject;
-    for (const [i, questionUpdate] of (
+    const subject: any = { ...subjectUpdate };
```

Comment:
avoid `any`...is what you want here either `Subject` or at worst `Partial<Subject>`?

```typescript
const subject: Subject = { 
  ...subjectUpdate ,
  _id: idOrNew(subjectUpdate._id),
  questions = subjectUpdate.questions || []
};

Reply:
subject and subjectUpdate are not the same type though (one is Subject and one is SubjectInputType) so typescript complains which is why I have it as any

Reply:
Basically everything is the same except for questions. subject should have questions be type { question: id, category: id, topics: id[] } but subjectUpdate is type { question: Question, category: Category, topics: Topic[] }